CMakeLists.txt

```diff
@@ -26,34 +26,47 @@
 #   cmake -G "Visual Studio 10 2010" ..
 #   msbuild /p:Configuration=Release ALL_BUILD.vcxproj
 #
+# Generic Invocation:
+#   cmake -E make_directory buildprod
+#   cd build
+#   cmake -G GENERATOR -DCMAKE_BUILD_TYPE=Release ..
+#   cmake --build . --target SOME_TARGET --config Release
+#
 # More information:
 #   cmake --help
 # 
 #   CMake is available at https://cmake.org/download/
 # 
 
-cmake_minimum_required (VERSION 2.6)
+cmake_minimum_required (VERSION 2.8.8)
+
+project(EPANET)
 
 SET(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
 SET(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
+SET(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-project(EPANET)
-
-IF(APPLE)
+IF (APPLE)
   SET(CMAKE_INSTALL_NAME_DIR @executable_path)
   SET(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
-ENDIF(APPLE)
+ENDIF (APPLE)
 
-# the library
+IF (MSVC)
+  set(CMAKE_C_FLAGS_RELEASE "/GL")
+  add_definitions(-D_CRT_SECURE_NO_DEPRECATE)
+ENDIF (MSVC)
+
+# create object library for reuse in other targets
 include_directories(include)
-file(GLOB EPANET_SOURCES src/*.c)
-file(GLOB EPANET_HEADERS src/*.h)
-file(GLOB EPANET_DATA src/*.dat)
-set(EPANET_API_HEADER "include/epanet2.h")
```

Comment:
these lines are important - or at the very least, the one that incorporates `epanet2.h` - otherwise the api header file won't be available in the project file generated by cmake

Reply:
Yes, I understand. I guess our changes are conflicting. I was experimenting with cmake trying to figure out how to avoid building everything for each target. 

So, if I set the `EPANET_API_HEADER` variable will that work for you?